pkg/vere/main.c

```diff
@@ -1726,6 +1727,165 @@ _cw_prep(c3_i argc, c3_c* argv[])
   u3_Host.ops_u.tem = c3y;
 }
 
+/* _cw_chop(): truncate event log
+*/
+static void
+_cw_chop(c3_i argc, c3_c* argv[])
+{
+  c3_i ch_i, lid_i;
+  c3_w arg_w;
+
+  static struct option lop_u[] = {
+    { "loom", required_argument, NULL, c3__loom },
+    { NULL, 0, NULL, 0 }
+  };
+
+  u3_Host.dir_c = _main_pier_run(argv[0]);
+
+  while ( -1 != (ch_i=getopt_long(argc, argv, "", lop_u, &lid_i)) ) {
+    switch ( ch_i ) {
+      case c3__loom: {
+        c3_w lom_w;
+        c3_o res_o = _main_readw(optarg, u3a_bits + 3, &lom_w);
+        if ( (c3n == res_o) || (lom_w < 20) ) {
+          fprintf(stderr, "error: --loom must be >= 20 and <= %u\r\n", u3a_bits + 2);
+          exit(1);
+        }
+        u3_Host.ops_u.lom_y = lom_w;
+      } break;
+
+      case '?': {
+        fprintf(stderr, "invalid argument\r\n");
+        exit(1);
+      } break;
+    }
+  }
+
+  //  argv[optind] is always "chop"
+  //
+
+  if ( !u3_Host.dir_c ) {
+    if ( optind + 1 < argc ) {
+      u3_Host.dir_c = argv[optind + 1];
+    }
+    else {
+      fprintf(stderr, "invalid command, pier required\r\n");
+      exit(1);
+    }
+
+    optind++;
+  }
+
+  if ( optind + 1 != argc ) {
+    fprintf(stderr, "invalid command\r\n");
+    exit(1);
+  }
+
+  // gracefully shutdown the pier if it's running
+  u3_disk* old_u = _cw_disk_init(u3_Host.dir_c);
+
+  // note: this include patch applications (if any)
+  u3m_boot(u3_Host.dir_c, (size_t)1 << u3_Host.ops_u.lom_y);
+
+  // check if there's a *current* snapshot
+  if ( old_u->dun_d != u3A->eve_d ) {
+    fprintf(stderr, "chop: error: snapshot is out of date, please "
+                    "start/shutdown your pier gracefully first\r\n");
+    fprintf(stderr, "chop: eve_d: %" PRIu64 ", dun_d: %" PRIu64 "\r\n", u3A->eve_d, old_u->dun_d);
+    exit(1);
+  }
+
+  u3e_backup(c3y);  //  backup current snapshot
```

Comment:
We can only continue here if `u3e_backup()` succeeds. It's currently a `void` function, with no indication of error (other than printfs). It'll need to return a status flag (at least success/failure), and we'll have to fail here if the backup was unsuccessful.

Reply:
Thank you, I should've caught this. 

Related: any more thoughts on whether or not we're going to be happy overwriting the existing backup snapshot?

Reply:
We don't really have better options right now. The way to resolve this is the epoch system.

Reply:
Update: Instead of having to decide whether or not to overwrite the backup snapshot (in `.urb/bhk`) if it exists, I'm just going to make a backup of the backup snapshot if it exists. It'll go in the `.urb/log/chop/bhk` folder.

Reply:
That won't work, you absolutely need the latest (current!) snapshot.

Reply:
Ok so we will go ahead with the original implementation: backup snapshot in (in `.urb/bhk`) will simply be overwritten. The real solution to this is the epoch system.